NuGetUpload/Services/PackageUploadService.cs

```diff
@@ -83,94 +82,116 @@ private async IAsyncEnumerable<string> ExtractZip(string zipFilePath, string bas
+                1; // Package push
 
             var stepSize = 100.0 / totalSteps;
-            var progress = 0.0;
+            var progressValue = 0.0;
 
-            (double, string) Step(string msg, double size = -1)
+            void Step(string msg, double? size = null)
             {
-                progress += size >= 0 ? size : stepSize;
-                return (progress, msg);
+                progressValue += size ?? stepSize;
+                progress?.Report((progressValue, msg));
             }
 
-            using var randomFolder = new TemporaryFolder(paths.TempUploads);
-            logger.LogInformation("Upload path: {}", randomFolder.Path);
-            var uploads = new List<string>();
-            foreach (var browserFile in files)
+            using var temporaryFolder = new TemporaryFolder(_pathsOptions.TempUploads);
+            _logger.LogDebug("Upload path: {Path}", temporaryFolder.Path);
+
+            var filePaths = new List<string>();
+            foreach (var file in files)
             {
-                var uploadMsg = $"Uploading {browserFile.Name}";
-                yield return Step(uploadMsg, 0);
-                var filePath = Path.Combine(randomFolder.Path, browserFile.Name);
-                var totalSize = (double)browserFile.Size;
-                await foreach (var read in Upload(browserFile, filePath))
-                    yield return Step(uploadMsg, read / totalSize * stepSize);
-
-                if (zipContentTypes.Contains(browserFile.ContentType))
+                var uploadMessage = $"Uploading {file.Name}";
+                Step(uploadMessage, 0);
+
+                var filePath = Path.Combine(temporaryFolder.Path, file.Name);
+                var totalSize = (double)file.Size;
+                await using var stream = file.OpenReadStream(MaxFileSize);
+                await foreach (var read in SaveToFile(stream, filePath))
+                    Step(uploadMessage, read / totalSize * stepSize);
+
+                if (_zipContentTypes.Contains(file.ContentType))
                 {
-                    yield return Step($"Extracting ZIP {browserFile.Name}", 0);
-                    await foreach (var s in ExtractZip(filePath, randomFolder.Path, uploads))
+                    Step($"Extracting ZIP {file.Name}", 0);
+                    await foreach (var s in ExtractZip(filePath, temporaryFolder.Path, filePaths))
                         Step(s, 0);
                 }
                 else
                 {
-                    uploads.Add(filePath);
+                    filePaths.Add(filePath);
                 }
             }
 
-            var allowedAssemblies =
-                new HashSet<string>(info.AllowedAssemblies, StringComparer.InvariantCultureIgnoreCase);
-            foreach (var filePath in uploads)
+            var allowedAssemblies = info.AllowedAssemblies == null ? null : new HashSet<string>(info.AllowedAssemblies, StringComparer.InvariantCultureIgnoreCase);
```

Comment:
Assembly allow list was added to prevent all packages always from uploading DLLs that were not agreed upon. As such, Allowing the assembly list to be `null` is not really an option, as I will always request package uploaders to state all assemblies they are going to upload to the service we host.

This can be technically reverted back to the list always being present. Though feel free to say if you don't want to, I suppose this can be in here. That being said, I certainly won't be providing this as an option to package uploaders 😅

Reply:
It doesn't really hurt to have more options in the config? Also it made testing the uploader a bit easier